modules/arrow/src/geoarrow/convert-geoarrow-to-binary-geometry.ts

```diff
@@ -271,47 +275,61 @@ export function getTriangleIndices(
   primitivePolygonIndices: Int32Array,
   flatCoordinateArray: Float64Array,
   nDim: number
-): Uint32Array {
-  let primitiveIndex = 0;
-  const triangles: number[] = [];
-  // loop polygonIndices to get triangles
-  for (let i = 0; i < polygonIndices.length - 1; i++) {
-    const startIdx = polygonIndices[i];
-    const endIdx = polygonIndices[i + 1];
-    // get subarray of flatCoordinateArray
-    const slicedFlatCoords = flatCoordinateArray.subarray(startIdx * nDim, endIdx * nDim);
-    // get holeIndices for earcut
-    const holeIndices: number[] = [];
-    while (primitivePolygonIndices[primitiveIndex] < endIdx) {
-      if (primitivePolygonIndices[primitiveIndex] > startIdx) {
-        holeIndices.push(primitivePolygonIndices[primitiveIndex] - startIdx);
+): Uint32Array | null {
+  try {
+    let primitiveIndex = 0;
+    const triangles: number[] = [];
+    // loop polygonIndices to get triangles
+    for (let i = 0; i < polygonIndices.length - 1; i++) {
+      const startIdx = polygonIndices[i];
+      const endIdx = polygonIndices[i + 1];
+      // get subarray of flatCoordinateArray
+      const slicedFlatCoords = flatCoordinateArray.subarray(startIdx * nDim, endIdx * nDim);
+      // get holeIndices for earcut
+      const holeIndices: number[] = [];
+      while (primitivePolygonIndices[primitiveIndex] < endIdx) {
+        if (primitivePolygonIndices[primitiveIndex] > startIdx) {
+          holeIndices.push(primitivePolygonIndices[primitiveIndex] - startIdx);
+        }
+        primitiveIndex++;
+      }
+      const triangleIndices = earcut(
+        slicedFlatCoords,
+        holeIndices.length > 0 ? holeIndices : undefined,
+        nDim
+      );
+      if (triangleIndices.length === 0) {
+        throw Error('can not tesselate invalid polygon');
+      }
+      for (let j = 0; j < triangleIndices.length; j++) {
+        triangles.push(triangleIndices[j] + startIdx);
       }
-      primitiveIndex++;
     }
-    const triangleIndices = earcut(
-      slicedFlatCoords,
-      holeIndices.length > 0 ? holeIndices : undefined,
-      nDim
-    );
-    for (let j = 0; j < triangleIndices.length; j++) {
-      triangles.push(triangleIndices[j] + startIdx);
+    // convert traingles to Uint32Array
+    const trianglesUint32 = new Uint32Array(triangles.length);
+    for (let i = 0; i < triangles.length; i++) {
+      trianglesUint32[i] = triangles[i];
     }
+    return trianglesUint32;
+  } catch (error) {
+    // there is an expection when tesselating invalid polygon, e.g. polygon with self-intersection
```

Comment:
Nit: Silently swallowing errors can make code hard to debug. We could add a console.log, but since this is in a loop, we could get a cascade, unless we use something like probe.gl which prevents cascades. So maybe just add a comment like `// TODO - add logging`?